final-project/scripts/main.js

```diff
@@ -1,10 +1,168 @@
+  readButton[0].addEventListener('click', function (event) {
+    window.location.replace(blogData.articles[currentBlog].url);
+  });
+  var image = document.getElementsByClassName("blog-section__content-image");
+  image[0].src = blogData.articles[currentBlog].images.desktop;
+}
+
+function addButtonListenersBlog() {
+  var arrowsBlogSlider = document.querySelector(".blog-section__arrows");
+  arrowsBlogSlider.addEventListener("click", updateVisibleBlog, false);
+  var buttonsBlogSlider = document.querySelector(".blog-section__slider");
+  buttonsBlogSlider.addEventListener("click", updateVisibleBlog, false);
+}
+
+function updateVisibleBlog(e) {
+  var length = Object.keys(blogData.articles).length;
+  if (e.target !== e.currentTarget) {
+    var clickedBlog = e.target.id;
+    if(clickedBlog === 'right') currentBlog += 1;
+    else if(clickedBlog === 'left') currentBlog -= 1;
+    else currentBlog = clickedBlog;
+    if (currentBlog >= length) currentBlog = 0;
+    if (currentBlog < 0) currentBlog = length-1;
+    updateSliderColor(currentBlog);
+    modifyBlogData(currentBlog);
+  }
+  e.stopPropagation();
+}
+
+function updateSliderColor(currentSlider) {
+  var buttons = document.getElementsByClassName("blog-section__slider-btn");
+  for (var i = 0; i < buttons.length; i++) {
+    buttons[i].style.backgroundColor = "#D1D6E3";
+  }
+  buttons[currentSlider].style.backgroundColor = "#5283FF";
+}
+
+function addButtonListenersCustomer(){
+  var buttonsParent = document.querySelector(".customers-section__slider");
+  buttonsParent.addEventListener("click", updateCurrentSlide, false);
+}
+
+function updateCurrentSlide(e) {
+  if (e.target !== e.currentTarget) {
+    var clickedCustomer = e.target.id;
+    currentCustomer(clickedCustomer);
+  }
+  e.stopPropagation();
+}
+
+
+function mediaQueries() {
+  var mobile = window.matchMedia("(max-width: 400px)");
+  screenWidthChangeMobile(mobile);
+  mobile.addListener(screenWidthChangeMobile);
+
+  var tabletPortrait = window.matchMedia("(min-width: 401px) and (max-width: 768px)");
+  screenWidthChangeTabletPortrait(tabletPortrait);
+  tabletPortrait.addListener(screenWidthChangeTabletPortrait);
+
+  var tabletLandscape = window.matchMedia("(min-width: 769px) and (max-width: 1024px)");
+  screenWidthChangeTabletLandscape(tabletLandscape);
+  tabletLandscape.addListener(screenWidthChangeTabletLandscape);
+
+  var desktop = window.matchMedia("(min-width: 1025px)");
+  screenWidthChangeDesktop(desktop);
+  desktop.addListener(screenWidthChangeDesktop);
+}
+
+function screenWidthChangeDesktop(desktop){
+  if(desktop.matches){
+    var x = document.getElementsByClassName("customers-section__images");
+    for (var i = 0; i < x.length; i++) {
+      x[i].style.display = "block";
+    }
+    var image = document.getElementsByClassName("blog-section__content-image");
+    image[0].src = blogData.articles[currentBlog].images.desktop;
```

Comment:
When you manage responsive images in this way you'll have problems when users don't have js turned on, I'd recommend to use HTML native responsive images instead https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images